File: Search/RecursiveLinearSearch.py
def recursiveLinearSearch(list, target, i=0):
  if list[i] == target:
    return i
  if list[len(list)-i-1] == target:
    return (len(list)-i-1)
  return recursiveLinearSearch(list, target, i+1)


myList = [0, 5, 8, 1, 763, 2, 9, 12]
print(myList)
print(recursiveLinearSearch(myList, 12))

File: Original/Search/RecursiveLinearSearch.java
import java.util.*;

public class RecursiveLinearSearch {

  public static int RecursiveLinSearch(List<Integer> l, int target, int index) {
    if (index >= l.size()) {
      return -1;
    }
    if (target == l.get(index)) {
      return index;
    }

    return RecursiveLinSearch(l, target, index + 1);
  }

  public static void main(String args[]) {
    List<Integer> l = new ArrayList<Integer>();

    int r = (int) (Math.random() * (40));
    int target = (int) (Math.random() * (r - 1));
    for (int i = 0; i < r; i++) {
      l.add(i);
    }
    System.out.println(l);
    System.out.println("Target: " + target + " at position: " + RecursiveLinSearch(l, target, 0));
  }
}
